Chore:(Docs) Complements the documentation with MDX

jonniebigodes opened the pull request:
With this pull request, the documentation is expanded to include examples for MDX. 

Addresses and closes #6725 and #15648.

What was done:
- Created snippets for React, Vue, Angular, Svelte.
- Adjusted the docs accordingly.
- Moved the placement of snippets to avoid confusion.
- Fixed a couple of items with current snippets

There's a small caveat to this pull request, as the args composition snippets were not adjusted and is something that can be followed up with another pull request. @shilman I'll follow up on this matter so that we can come up with a good approach and we'll go from there.


Feel free to provide feedback

jonniebigodes committed:
complements the docs with MDX

jonniebigodes commented:
@winkerVSbecks care to take a look at this pull request and let me know if you spot anything that needs fixing?

winkerVSbecks commented:
LGTM

shilman commented on docs/snippets/angular/badge-story.mdx.mdx:
```diff
@@ -0,0 +1,58 @@
+  status: 'positive',
+  label: 'Positive'
+}}>
+{Template.bind({})}
```
```suggestion
  {Template.bind({})}
```

jonniebigodes replied:
Fixed with the latest commits

shilman commented on docs/snippets/angular/button-story-component-decorator.mdx.mdx:
```diff
@@ -0,0 +1,34 @@
+    moduleMetadata({
+      declarations: [ParentComponent],
+    }),
+     componentWrapperDecorator(ParentComponent),
```
```suggestion
    componentWrapperDecorator(ParentComponent),
```

jonniebigodes replied:
Fixed with the latest commits

shilman commented on docs/snippets/angular/checkbox-story.mdx.mdx:
```diff
@@ -0,0 +1,37 @@
+    label: 'Secondary', 
+    checked: true, 
+    appearance: 'secondary'
+  }}>
```
```suggestion
    }}>
```

jonniebigodes replied:
Fixed with the latest commits

shilman reviewed (approved):
Looks good! Lots of inconsistent indentation here -- I'm sure I only caught a fraction.

jonniebigodes committed:
adjustments to the mdx snippets per feedback

jonniebigodes committed:
adds angular fixed snippet

jonniebigodes commented:
With the latest commits, I've polished and updated the snippets based on feedback and ran through the other mdx examples currently documented and adjusted them to include missing imports, and apply some consistency to them

shilman reviewed (commented):
I think our prettier setup uses single quotes for code but double quotes for JSX attributes. I don't particularly care one way or the other, but we should probably be consistent. Even better, is there some way to get prettier to run on the MDX snippets inside the MDX files?

jonniebigodes commented:
I'll adjust accordingly! I saw that some of the mdx snippets were using single quotes, for example [here](https://storybook.js.org/docs/react/essentials/actions#gatsby-focus-wrapper) and some other places so I tried a unified approach to this. But I can adjust accordingly based on what you've mentioned. I'll go through them and push a commit in a bit. Regarding your question:
>Even better, is there some way to get prettier to run on the MDX snippets inside the MDX files?

I've had mixed results where it formatted the code properly, other times it broke the example/snippet formatting altogether.

Probably some rule or some misconfiguration that is missing or not being applied properly.

jonniebigodes committed:
Normalized the MDX snippets all around

jonniebigodes commented:
Just ran through the entire mdx snippets and applied the same formatting for all of them.
